qt/update_dialog.cpp

```diff
@@ -402,8 +407,43 @@ namespace qt
     m_tree->setSortingEnabled(false);
     m_tree->clear();
 
+    vector<TCountryId> toShow;
+
+    if (!filter.empty())
+    {
+      condition_variable cv;
+      mutex mu;
+      bool done = false;
+
+      DownloaderSearchParams params;
+      params.m_query = filter;
+      params.m_inputLocale = "en";
+      params.m_onResults = [&](DownloaderSearchResults const & results)
+      {
+        lock_guard<mutex> lock(mu);
+
+        for (auto const & res : results.m_results)
+          toShow.push_back(res.m_countryId);
+
+        if (results.m_endMarker)
+        {
+          done = true;
+          cv.notify_one();
+        }
+      };
+
+      m_framework.SearchInDownloader(params);
+
+      {
```

Comment:
I generally don't like the idea of blocking call to Search. Could you please investigate - is it possible to fill results asynchronously?

Reply:
done